include/openssl/crypto.h.in

```diff
@@ -93,14 +93,21 @@ int CRYPTO_atomic_load(uint64_t *val, uint64_t *ret, CRYPTO_RWLOCK *lock);
 /* No longer needed, so this is a no-op */
 #define OPENSSL_malloc_init() while(0) continue
 
+void *ERR_raise_malloc_failure(void *ptr, size_t num,
+                               const char *file, int line, const char *func);
 # define OPENSSL_malloc(num) \
-        CRYPTO_malloc(num, OPENSSL_FILE, OPENSSL_LINE)
+    ERR_raise_malloc_failure(CRYPTO_malloc(num, OPENSSL_FILE, OPENSSL_LINE), \
+                             num, OPENSSL_FILE, OPENSSL_LINE, OPENSSL_FUNC)
```

Comment:
You cannot duplicate num here, that would be an API breakage of the OPENSSL_malloc() macro. Unfortunately that means you cannot report the size of the allocation that failed but that is not a regression of functionality from existing code. So I do not think this is a problem.